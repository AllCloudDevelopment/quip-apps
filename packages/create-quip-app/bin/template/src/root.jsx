import quip from "quip";
import App from "./App.jsx";

quip.elements.initialize({
    initializationCallback: function(rootNode) {
        ReactDOM.render(<App />, rootNode);
    },
});
