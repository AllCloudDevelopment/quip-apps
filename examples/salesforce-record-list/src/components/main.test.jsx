// Copyright 2019 Quip

import React from "react";
import {shallow, mount} from "enzyme";
import quip from "quip-apps-api";
import {RootEntity} from "../model/root";
import {ListMenu} from "../menus";

import Main from "./main";

class TestAuth extends quip.apps.OAuth2 {}

describe("Main Component", () => {
    test("Logged Out View", () => {
        const rootRecord = new RootEntity();
        const listMenu = new ListMenu({});
        const wrapper = shallow(
            <Main
                rootRecord={rootRecord}
                listMenu={listMenu}
                isCreation={true}/>);
        expect(wrapper.find(".loginContext")).toHaveLength(1);
        expect(wrapper.find(".loginContext")).toMatchSnapshot();
        expect(wrapper.find(".openPicker")).toHaveLength(1);
        expect(wrapper.find(".openPicker")).toMatchInlineSnapshot(`
            <div
              className="openPicker"
            >
              <Button
                disabled={false}
                text="Connect to Salesforce"
              />
            </div>
        `);
    });

    test("Empty List", () => {
        const rootRecord = new RootEntity();
        const listMenu = new ListMenu({});
        const getData_ = rootRecord.getData;
        rootRecord.getData = jest.fn().mockImplementation(function(...args) {
            const data = getData_.call(this, ...args);
            data.selectedList.rows = [];
            data.selectedList.isEmpty = true;
            return data;
        });
        const wrapper = mount(
            <Main
                rootRecord={rootRecord}
                listMenu={listMenu}
                isCreation={true}/>);
        expect(wrapper.find(".emptyMessage")).toHaveLength(1);
        expect(wrapper.find(".emptyMessage")).toMatchInlineSnapshot(`
            <div
              className="emptyMessage"
            >
              <h2>
                No Records Found
              </h2>
            </div>
        `);
    });

    test("creationUrl triggers selectList with the correct args", () => {
        const rootRecord = new RootEntity();
        const listMenu = new ListMenu({});
        const creationUrl =
            "https://gus.lightning.force.com/lightning/o/ADM_Work__c/list?filterName=00BB0000000LaOaMAK";
        const instance = <Main
            rootRecord={rootRecord}
            listMenu={listMenu}
            isCreation={true}
            creationUrl={creationUrl}/>;

        const willMount = Main.prototype.componentDidMount;
        let spy;
        jest.spyOn(Main.prototype, "componentDidMount").mockImplementationOnce(
            function() {
                spy = jest.spyOn(this, "onSelectList_");
                willMount.apply(this, arguments);
            });
        shallow(instance);
        expect(spy).toBeDefined();
        expect(spy).toHaveBeenCalledWith("00BB0000000LaOaMAK", "ADM_Work__c");
    });

    test("initOptions triggers selectList with the correct args", () => {
        const rootRecord = new RootEntity();
        rootRecord.setAuth(new TestAuth());
        const listMenu = new ListMenu({});
        const instance = <Main
            rootRecord={rootRecord}
            listMenu={listMenu}
            isCreation={false}
            initOptionsJson={
                '{"list_id": "00BB0000000LaOaMAK", "object_type": "ADM_Work__c"}'
            }/>;

        const willMount = Main.prototype.componentDidMount;
        let spy;
        jest.spyOn(Main.prototype, "componentDidMount").mockImplementationOnce(
            function() {
                spy = jest.spyOn(this, "onSelectList_");
                willMount.apply(this, arguments);
            });
        shallow(instance);
        expect(spy).toBeDefined();
        expect(spy).toHaveBeenCalledWith("00BB0000000LaOaMAK", "ADM_Work__c");
    });
});
