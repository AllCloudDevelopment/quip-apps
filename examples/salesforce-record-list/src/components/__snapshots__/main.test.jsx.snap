// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component Logged Out View 1`] = `
<div
  className="loginContext"
>
  <span>
    Trouble connecting? Make sure your Quip and Salesforce admin has 
    <a
      href="https://www.quipsupport.com/hc/en-us/articles/360022156351-Quip-Salesforce-Integration-Summary"
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      already set up the integration
    </a>
    , and that your browser isn't blocking the login popup.
  </span>
</div>
`;
